test-data/unit/check-incremental.test

```diff
@@ -4283,3 +4283,31 @@ def foo() -> None:
 [out]
 [out2]
 tmp/main.py:2: error: "int" has no attribute "foo"
+
+[case testIncrementalBustedFineGrainedCache1]
+# flags: --cache-fine-grained
+import a
+import b
+[file a.py]
+[file b.py]
+-- This is a heinous hack, but we simulate having a invalid cache by clobbering
+-- the proto deps file with something with a hash mismatch.
+[file ../.mypy_cache/3.6/@proto_deps.meta.json.2]
+{"__main__": "00000000000000000000000000000000", "a": "d41d8cd98f00b204e9800998ecf8427e", "b": "d41d8cd98f00b204e9800998ecf8427e", "builtins": "3d22863e34b3d846306e1e76ebc82407"}
+[file b.py.2]
+# uh
+-- Every file should get reloaded, since the cache was invalidated
+[stale a, b, builtins]
+[rechecked a, b, builtins]
+
+[case testIncrementalBustedFineGrainedCache2]
```

Comment:
Maybe also add a test where everything works for `--cache-fine-grained` and `--incremental` (i.e. flag is used on both runs, we don't touch proto_deps, and then check that only relevant files are rechecked)?

Reply:
Done